pkg/services/provisioning/alerting/rules/types.go

```diff
@@ -0,0 +1,210 @@
+package rules
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/grafana/grafana/pkg/services/ngalert/models"
+	"github.com/grafana/grafana/pkg/services/provisioning/values"
+)
+
+type configVersion struct {
+	APIVersion values.Int64Value `json:"apiVersion" yaml:"apiVersion"`
+}
+
+type RuleFile struct {
+	configVersion
+	Groups      []AlertRuleGroup
+	DeleteRules []RuleDelete
+}
+
+type RuleFileV1 struct {
+	configVersion
+	Groups      []AlertRuleGroupV1 `json:"groups" yaml:"groups"`
+	DeleteRules []RuleDeleteV1     `json:"deleteRules" yaml:"deleteRules"`
+}
+
+func (ruleFileV1 *RuleFileV1) MapToModel() (RuleFile, error) {
+	ruleFile := RuleFile{}
+	ruleFile.configVersion = ruleFileV1.configVersion
+	for _, groupV1 := range ruleFileV1.Groups {
+		group, err := groupV1.mapToModel()
+		if err != nil {
+			return RuleFile{}, err
+		}
+		ruleFile.Groups = append(ruleFile.Groups, group)
+	}
+	for _, ruleDeleteV1 := range ruleFileV1.DeleteRules {
+		orgID := ruleDeleteV1.OrgID.Value()
+		if orgID < 1 {
+			orgID = 1
+		}
+		ruleDelete := RuleDelete{
+			UID:   ruleDeleteV1.UID.Value(),
+			OrgID: orgID,
+		}
+		ruleFile.DeleteRules = append(ruleFile.DeleteRules, ruleDelete)
+	}
+	return ruleFile, nil
+}
+
+type RuleDelete struct {
+	UID   string
+	OrgID int64
+}
+
+type RuleDeleteV1 struct {
+	UID   values.StringValue `json:"uid" yaml:"uid"`
+	OrgID values.Int64Value  `json:"orgId" yaml:"orgId"`
+}
+
+type AlertRuleGroupV1 struct {
+	OrgID    values.Int64Value  `json:"orgId" yaml:"orgId"`
+	Name     values.StringValue `json:"name" yaml:"name"`
+	Folder   values.StringValue `json:"folder" yaml:"folder"`
```

Comment:
This has to be a UID right? Should we call this `FolderUID` to reduce confusion? I could see users trying to put a folder name in here and getting confused why it isn't working.

Reply:
I realized this was wrong after the fact. It's folder name.